sammashar56/mtush_backend
# #168060537 User Authentication

#### What does this PR do?

This pr handles authentication of the user

#### Description of tasks to be accomplished
- user should be able to login
- user should be able to update profile
- user should be able to signup
- user should be able to reset password
- user should be able to verify account

#### How can this be manually tested

for login 
*Endpoint*: `localhost:3040/mtush/v1/in`
```json
{
	"email": "erick@werick.codes",
	"password": "Wachira254"
}

```

for signup
*Endpoint*: `localhost:3040/mtush/v1/join`
```json
{
	"email": "ewachira254@gmail.com",
	"password": "Wachira254",
	"name": "Erick Wachira",
	"confirm": "Wachira254",
	"username": "Wachira",
        "role":<default is User, Shop, Admin>
}
```

for verification
*Endpoint*: `http://localhost:3040/mtush/v1/email-verify?token=<token from email>`

`use the link sent to your email`

for password reset request
*Endpoint*: `localhost:3040/mtush/v1/password-reset/email`

```json

{
	"email":"ewachira254@gmail.com"
}

```

for the password reset process
*Endpoint*: `localhost:3040/mtush/v1/password-reset/<token from email>`
`use the link from the email to send request`
```json
{
	"password":"@Wachira254",
	"confirm": "@Wachira254"
}
```

#### Related Pivotal Tracker story?

[#168060537](https://www.pivotaltracker.com/story/show/168060537)
